modules/market/src/test/java/com/opengamma/strata/market/curve/InterpolatedNodalCurveTest.java

```diff
@@ -146,58 +159,51 @@ public void test_withYValues_badSize() {
   }
 
   //-------------------------------------------------------------------------
-  public void test_withNode_atStart_noMetadata() {
+  public void test_withNode_atStart_withMetadata() {
     InterpolatedNodalCurve base = InterpolatedNodalCurve.of(METADATA_ENTRIES, XVALUES, YVALUES, INTERPOLATOR);
-    InterpolatedNodalCurve test = base.withNode(0, 0.5d, 4d);
-    DoubleArray x = DoubleArray.of(0.5d, 1d, 2d, 3d);
-    DoubleArray y = DoubleArray.of(4d, 5d, 7d, 8d);
+    LabelDateParameterMetadata item = LabelDateParameterMetadata.of(date(2015, 6, 30), TNR_1Y);
+    InterpolatedNodalCurve test = base.withNode(0.5d, 4d, item);
+    List<ParameterMetadata> list = new ArrayList<>();
+    list.add(item);
+    list.addAll(ParameterMetadata.listOfEmpty(SIZE));
     assertThat(test.getName()).isEqualTo(CURVE_NAME);
     assertThat(test.getParameterCount()).isEqualTo(SIZE + 1);
-    assertThat(test.getMetadata()).isEqualTo(METADATA);
-    assertThat(test.getXValues()).isEqualTo(x);
-    assertThat(test.getYValues()).isEqualTo(y);
+    assertThat(test.getMetadata()).isEqualTo(METADATA_ENTRIES.withParameterMetadata(list));
+    assertThat(test.getXValues()).isEqualTo(DoubleArray.of(0.5d, 1d, 2d, 3d));
+    assertThat(test.getYValues()).isEqualTo(DoubleArray.of(4d, 5d, 7d, 8d));
   }
 
-  public void test_withNode_atEnd_noMetadata() {
-    InterpolatedNodalCurve base = InterpolatedNodalCurve.of(METADATA_ENTRIES, XVALUES, YVALUES, INTERPOLATOR);
-    InterpolatedNodalCurve test = base.withNode(SIZE, 4d, 9d);
-    DoubleArray x = DoubleArray.of(1d, 2d, 3d, 4d);
-    DoubleArray y = DoubleArray.of(5d, 7d, 8d, 9d);
+  public void test_withNode_atEnd_withoutMetadata() {
+    InterpolatedNodalCurve base = InterpolatedNodalCurve.of(METADATA, XVALUES, YVALUES, INTERPOLATOR);
+    LabelDateParameterMetadata item = LabelDateParameterMetadata.of(date(2015, 6, 30), TNR_1Y);
+    InterpolatedNodalCurve test = base.withNode(0.5d, 4d, item);
     assertThat(test.getName()).isEqualTo(CURVE_NAME);
     assertThat(test.getParameterCount()).isEqualTo(SIZE + 1);
     assertThat(test.getMetadata()).isEqualTo(METADATA);
-    assertThat(test.getXValues()).isEqualTo(x);
-    assertThat(test.getYValues()).isEqualTo(y);
+    assertThat(test.getXValues()).isEqualTo(DoubleArray.of(0.5d, 1d, 2d, 3d));
+    assertThat(test.getYValues()).isEqualTo(DoubleArray.of(4d, 5d, 7d, 8d));
   }
 
-  public void test_withNode_atStart_metadata() {
+  public void test_withNode_replace_withMetadata() {
```

Comment:
There's no test for inserting a new node in the middle of the curve, only at the start. Is that worth having?